src/chainstate/burn/distribution.rs

```diff
@@ -186,19 +201,64 @@ impl BurnSamplePoint {
                 .map(|missed| (missed.txid.clone(), missed))
                 .collect();
 
-            let sunset_finished = if let Some(sunset_finished_at) = sunset_finished_at {
-                sunset_finished_at <= rel_block_height
+            let expected_index_opt = if rel_block_height > 0 {
+                // NOTE: expect_burnt refers to whether or not this block at `rel_block_height`
+                // consumes a UTXO from a PoB-only input (e.g. a prepare-phase commit or a post-sunset
+                // commit), or from a potentially-PoX input.  To determine this, we look at
+                // whether or not the _previous_ block-commit from rel_block_height is a
+                // PoB-only block-commit.  If it is, then the block-commit at `rel_block_height`
+                // would expect that its consumed UTXO occurs after a _single_ burn output,
+                // instead of `OUTPUTS_PER_COMMIT` PoX (or PoB) outputs.
+                let expect_burnt = burn_blocks[(rel_block_height - 1) as usize];
+                let expected_index = LeaderBlockCommitOp::expected_chained_utxo(expect_burnt);
+                Some(expected_index)
             } else {
-                false
+                None
             };
-            let expected_index = LeaderBlockCommitOp::expected_chained_utxo(sunset_finished);
+
             for linked_commit in commits_with_priors.iter_mut() {
                 let end = linked_commit.iter().rev().find_map(|o| o.as_ref()).unwrap(); // guaranteed to be at least 1 non-none entry
 
-                // check that the commit is using the right output index
-                if end.op.spent_output() != expected_index {
```

Comment:
Previously, this branch checked that `end.op` spent the correct output from the "current" block. Now, this code checks that "current" spends the correct output from it's prior block. I think this shift causes a check to be missed between the latest commit in the window and the second-latest commit in the window (as well as requiring the introduction of an option type for the check, which shouldn't be necessary if we're always checking from the end to the "current").

Reply:
Good catch!  Fixed in 9c6035c